proto/cachepubsub.proto

```diff
@@ -0,0 +1,126 @@
+//   don't bother handling them! You might still want to log them just in case ;-)
+service Pubsub {
+  // Publish a message to a topic.
+  //
+  // If a topic has no subscribers, then the effect of Publish MAY be either of:
+  // * It is dropped and the topic is nonexistent.
+  // * It is accepted to the topic as the next message.
+  //
+  // Publish() does not wait for subscribers to accept. It returns Ok upon accepting
+  // the topic value. It also returns Ok if there are no subscribers and the value
+  // happens to be dropped. Publish() can not guarantee delivery in theory but in
+  // practice it should almost always deliver to subscribers.
+  //
+  // REQUIRES HEADER authorization: Momento auth token
+  rpc Publish(_PublishRequest) returns (_Empty);
+
+  // Subscribe to notifications from a topic.
+  //
+  // You will receive a stream of values and (hopefully occasional) discontinuities.
+  // Values will appear as copies of the payloads you Publish() to the topic.
+  //
+  // REQUIRES HEADER authorization: Momento auth token
+  rpc Subscribe(_SubscriptionRequest) returns (stream _SubscriptionItem);
+}
+
+message _Empty {}
+
+// A value to publish through a topic.
+message _PublishRequest {
+  // Cache namespace for the topic to which you want to send the value.
+  string cache_name = 1;
+  // The literal topic name to which you want to send the value.
+  string topic = 2;
+  // The value you want to send to the topic. All current subscribers will receive
+  // this, should the whims of the Internet prove merciful.
+  _TopicValue value = 3;
+}
+
+// A description of how you want to subscribe to a topic.
+message _SubscriptionRequest {
+  // Cache namespace for the topic to which you want to subscribe.
+  string cache_name = 1;
+
+  // The literal topic name to which you want to subscribe.
+  string topic = 2;
+
+  // --> Providing this is not required. <--
+  //
+  // If provided, attempt to reconnect to the topic and replay messages starting from
+  // the provided sequence number. You may get a discontinuity if some (or all) of the
+  // messages are not available.
+  // If not provided (or 0), the subscription will begin with the latest messages.
+  uint64 resume_at_topic_sequence_number = 3;
+}
+
+// Possible message kinds from a topic. They can be items when they're from you, or
+// other kinds when we have something we think you might need to know about the
+// subscription's status.
+message _SubscriptionItem {
+  oneof kind {
+    // The subscription has yielded an item you previously published.
+    _TopicItem item = 1;
+    // Momento wants to let you know we detected some possible inconsistency at this
+    // point in the subscription stream.
+    //
+    // A lack of a discontinuity does not mean the subscription is guaranteed to be
+    // strictly perfect, but the presence of a discontinuity is very likely to
+    _Discontinuity discontinuity = 2;
```

Comment:
do we proactively send down a discontinuity whenever we detect one? (probably mr2 who will detect these is my guess).



or do we do this only when the client provides us with `resume_at_topic_sequence_number`

Reply:
our backend can detect certain kinds of discontinuities while our frontend would know about others. Either can relay a discontinuity. Wherever you'd `log { msg("we lost some data") }` you can `emit(discontinuity(fromNum, toNum))` instead and give the stream more rich metadata about its quality. It'll be nice for monitoring